plugin/src/features/Compilation/index.tsx

```diff
@@ -36,45 +38,66 @@ export const Compilation = ({ setAccordian }: CompilationProps) => {
 
   const setContracts = useSetAtom(contractsAtom)
   const setSelectedContract = useSetAtom(selectedContractAtom)
+  const setDeployStatus = useSetAtom(deployStatusAtom)
 
   const { status, isCompiling } = useAtomValue(compilationAtom)
 
   const setCompileStatus = useSetAtom(compileStatusAtom)
   const setIsCompiling = useSetAtom(isCompilingAtom)
+  const [isCompilingProject, setIsCompilingProject] = useState(false)
   const setCompileErrorMessages = useSetAtom(compileErrorMessagesAtom)
+  const [isContractsFolderAvailable, setIsContractsFolderAvailable] = useState(true)
 
   const solidityVersion = useAtomValue(solidityVersionAtom)
 
-  const compilations = [
-    {
-      validation: isValidSolidity,
-      isLoading: isCompiling,
-      onClick: compile,
-      id: 1
-    }
-  ]
+  useEffect(() => {
+    remixClient.fileManager.readdir(`${currentWorkspacePath}/`).then((workspaceFiles: any) => {
+      if (!workspaceFiles || !workspaceFiles['contracts'] || workspaceFiles['contracts']['isDirectory'] === false) {
```

Comment:
That shall be possible

```suggestion
setIsContractsFolderAvailable(
  !!workspaceFiles?.contracts?.isDirectory
);
```

Reply:
makes sense.